androiddemo/build.gradle.kts

```diff
@@ -14,6 +15,13 @@ android {
         versionName = "1.0"
 
         testInstrumentationRunner = "androidx.test.runner.AndroidJUnitRunner"
+
+        // Default values for demo purposes
+        // You can manually change these values or set them via environment variables
+        buildConfigField("String", "DEMO_RPC_URL", "\"${System.getenv("DEMO_RPC_URL") ?: "http://example-node-url.com/rpc"}\"")
+        buildConfigField("String", "DEMO_ACCOUNT_ADDRESS", "\"${System.getenv("DEMO_ACCOUNT_ADDRESS") ?: "0x12345"}\"")
+        buildConfigField("String", "DEMO_PRIVATE_KEY", "\"${System.getenv("DEMO_PRIVATE_KEY") ?: "0x123"}\"")
+        buildConfigField("String", "DEMO_RECIPIENT_ACCOUNT_ADDRESS", "\"${System.getenv("DEMO_RECIPIENT_ACCOUNT_ADDRESS") ?: "0x789"}\"")
```

Comment:
> I'm afraid this would lead to some confusion, as the default values won't work

I assume you mean no demo functionalities could be used with the default values? This is true. 

But we can't **not set anything**. In we android demo we can't fetch variables directly: android emulator doesn't have access to the host environment. They need to be set during the build.

Of course, if you think the app should crash right away, instead of defaults we can use some placeholder value like `""`, and crash if we `buildConfigField` is `""` instead of an actual value. I'm not sure that would be much clearer than a toast that shows what's wrong:
<img width="248" alt="Screenshot 2023-10-24 at 16 01 46" src="https://github.com/software-mansion/starknet-jvm/assets/43750648/a88399dd-d860-42fc-b602-73bbe12033a3">

Reply:
> In that case these values should be set to some real values from devnet

I intended default values to be dummies, so the user can clearly get that they probably should be replaced.

> a disclaimer should be added that this demo works only with devnet with specific seed

Should we rely on devnet for this? Devnet is not exactly the most reliable right now. With testnet/integration, you can be sure it's the latest version, and we don't need to also instruct a newcomer on how to properly build devnet (still no releases) and how exactly to run it in order to access the demo.

Reply:
I'm not even sure we can run devnet on android. If we can it's probably not quite straightforward.

Reply:
> I intended default values to be dummies, so the user can clearly get that that they should be replaced.

This is non issue, if we use devnet values; if there's no devnet the bahiour stays the same as right now; if there is devnet (which we should outline somewhere in redme that the demo uses), the demo just works without any input from user. this is the way this should be done imho.

> Should we (at least currently) rely on devnet for this?
Devnet is not exactly the most reliable right now. With testnet/integration, you can be sure it's the latest version, and we don't need to also instruct a newcomer on how to properly build devnet (still no releases) and how exactly to run it in order to access the demo.

I agree that building is a pain in the ass right now, but it is what it is. It is still simpler, than figuring out how to create an integration/testnet node, create and prefund the account there, then deploy it. We can easily instruct users in readme how to get working srtup in 3 lines of code.

> I'm not even sure we access devnet from android device or even emulator. Also, the demo becomes quite limited (you can't later use starkscan to see more about demo's transactions).

If it can access any url, it can access devnet :+1: if someone knows what starkscan is and wants to use the demo with some real node, I don't think this user will have any problem with supplying the necessary info to the demo. It's the newcomers and newbies I worry about, and I think we should make the demo as easy as it gets - just follow these few steps in readme (clone devnet, build it, run it), and you have a working demo - no additional configuration.